hazelcast/src/main/java/com/hazelcast/partition/MigrationListener.java

```diff
@@ -19,31 +19,55 @@
 import java.util.EventListener;
 
 /**
- * MigrationListener provides the ability to listen to partition migration events.
+ * MigrationListener provides the ability to listen to
+ * partition migration process and events.
  *
  * @see Partition
  * @see PartitionService
+ * @see MigrationState
+ * @see ReplicaMigrationEvent
  */
 public interface MigrationListener extends EventListener {
 
     /**
-     * Invoked when a partition migration is started.
+     * Called when the migration process starts.
+     * A migration process consists of a group of partition
+     * replica migrations which are planned together.
+     * <p>
+     * When migration process is completed, {@link #migrationFinished(MigrationState)}
+     * is called.
      *
-     * @param migrationEvent the event for the started partition migration
+     * @param event Plan of the migration process
      */
-    void migrationStarted(MigrationEvent migrationEvent);
+    void migrationStarted(MigrationState event);
```

Comment:
Maybe it is better to name the parameter `state` instead of `event` and `result` (see `migrationFinished`)? So that all places of public API (`MigrationListener` methods, `ReplicaMigrationEvent`) refer to it in a similar manner.

Reply:
👍